test/custom_clojure_ttt/console_ui/console_ui_game_setup_test.clj

```diff
@@ -0,0 +1,81 @@
+(ns custom-clojure-ttt.console_ui.console_ui_game_setup_test
+  (:require [clojure.string :as str]
+            [clojure.test :refer [deftest testing is]]
+            [clojure-tic-tac-toe.default_valid_moves :as default_valid_moves]
+            [clojure-tic-tac-toe.default_winning_moves :as default_winning_moves]
+            [custom-clojure-ttt.console_ui.console_ui_computer_move :as ui_comp_move]
+            [custom-clojure-ttt.console_ui.console_ui_human_move :as ui_human_move]
+            [custom-clojure-ttt.console_ui.console_ui_game_setup :refer :all]
+            [custom-clojure-ttt.view_handler :as view_handler]))
+
+(defn- perform-setup-output []
+  (with-out-str
+    (with-in-str "h\n"
+      (perform-setup))))
+
+(deftest perform-setup-test
+  (testing "when the player chooses to play another person"
+    (is (= true
+           (str/includes?
+             (perform-setup-output)
+             "Tic Tac Toe"))
+        "it displays the introduction")
+    (is (= true
+           (str/includes?
+             (perform-setup-output)
+             "\"h\" to play"))
+        "it displays the game mode instructions")
+    (is (= true
+           (str/includes?
+             (perform-setup-output)
+             "chose to play another person"))
+        "it displays the 'playing person' message")
+    (with-out-str
+      (is (= :human
+             (with-in-str "h\n"
+               (perform-setup)))
+          "it returns the internal keyword for playing a person"))))
+
+(deftest get-valid-moves-test
+  (testing "when a side length is passed in"
+    (is (= default_valid_moves/valid-moves
+           (get-valid-moves :3))
+        "it returns the valid moves")))
+
+(deftest get-winning-moves-test
+  (testing "when the function is called"
+    (is (= default_winning_moves/winning-moves
+           (get-winning-moves :3))
```

Comment:
Let's say the `default_winning_moves/winning-moves` changed so that it's for a 4x4 now. This test would still pass & so therefore it's not really a useful test. In this case, it'd be better to test it against actual winning moves

Reply:
same with the valid-moves one.

Reply:
Both of these tests (`get-winning-moves-test` and `get-valid-moves-test`) are meant to just confirm part of the functionality of their respective function. They are being fully tested in `valid_move_handler` and `winning_move_handler`. However, since `console_ui_game_setup` wraps them, I just wanted to have a confirmation test for both.